components/layout_2020/flexbox/layout.rs

```diff
@@ -943,6 +943,9 @@ impl FlexLine<'_> {
                 //  it’s a min violation.”
                 for (item_and_target_main_size, frozen) in items() {
                     if violation(item_and_target_main_size) > Length::zero() {
+                        item_and_target_main_size
+                            .1
+                            .set(item_and_target_main_size.0.content_min_size.main);
```

Comment:
I'm not sure this is the proper logic, shouldn't the clamping happen unconditionally regardless of `total_violation`?

Maybe `total_violation` is positive because most items have a big min size, but there is an item with a negative violation that needs to be clamped to the max size?

Reply:
In practice this won't happen, beacause all items will either grow or shrink in one direction, and items with negative violation is inflexible and has already been frozen in a previous iteration.

Reply:
Oh, it is a separate step in line 840 that size all inflexible items so they won't participate in the later loop.

Reply:
OK I get it, non-frozen items with a violation with a different sign than `total_violation` don't need to clamp their target main size because they will stay non-frozen, and thus in the next step we will compute the free space using their base size instead of their target main size, and then we will overwrite their target main size.